src/seedu/addressbook/commands/Command.java

```diff
@@ -25,6 +25,14 @@ public Command(int targetIndex) {
 
     protected Command() {
     }
+    
+    /**
+     * Returns if the command mutates data
+     */
+    public boolean isMutating() {
+        return true;
+    }
+    //Note: Default behaviour is to return true as it is better to assume commands mutate data to prevent incorrectly not storing data
```

Comment:
This should be ideally put in the header comments allowing programmers who use this function to note the behaviour.